docs/index.rst

```diff
@@ -34,11 +32,21 @@ based on local resource availability.
 Getting Started
 ---------------
 
-1. `pip install total-perspective-vortex` into Galaxy's python virtual environment
+1. ``pip install total-perspective-vortex`` into Galaxy's python virtual environment
 2. Configure Galaxy to use TPV's dynamic destination rule
 3. Create the TPV job mapping yaml file, indicating job routing preferences
 4. Submit jobs as usual
 
+Standalone Installation
+-----------------------
+
+If you wish to install TPV outside of Galaxy's virtualenv (e.g. to use the ``tpv lint`` command locally or in a CI/CD
+pipeline), use the ``[cli]`` pip requirement specifier to make sure the necessary Galaxy dependency packages are also
+installed. **This should not be used in the Galaxy virtualenv**:
```

Comment:
The main use case would be to quickly check where, in production, the job would dispatch to right? Since tpv (including shell commands), will be installed and available in Galaxy's virtualenv, what if we assume, that dry-run will also generally be invoked from within that directory and automatically import stuff from lib? That way, we could also auto-discover the galaxy/job_config, reducing the command to:
`tpv eval my_tool_id`

For use cases outside of that, someone can still install REQS_CLI? This saves the hassle of maintaining yet another venv.

Reply:
I've been using it locally to test before deploying, but yeah it would also be used to live test on the server.

Most production installs don't use the standard in-source layout so it's harder to make assumptions about where everything is (e.g. `galaxy.yml` is not in `<cwd>/config/galaxy.yml`. And because Galaxy libs aren't installed in the venv, you have to have your `cwd` as the galaxy root, which is always a hassle.

We could read `$GALAXY_CONFIG_FILE` I suppose. Finding and reading the job conf is a little tricky now since it can be inline, YAML, or XML. And the galaxy root could be anywhere, although there is a gravity config option for it in `galaxy.yml`.

Anyway, there's a lot of searching and reading and parsing stuff that I am not sure if you want to add to TPV. Gravity already does this so I could maybe break that into a separate library (or somewhere in Galaxy itself) and TPV could use that?

Reply:
Fair point. So perhaps a commandline param could just point to galaxy.yml, which could be passed to the MockApp from Galaxy's integration test, which would discover the job_conf + all other configs. This is only relevant if/when we move towards a more accurate simulation.